src/Features/Core/Portable/IntroduceVariable/AbstractIntroduceVariableService.State.cs

```diff
@@ -229,14 +229,21 @@ private TExpressionSyntax GetExpressionUnderSpan(SyntaxTree tree, TextSpan textS
             }
 
             private bool CanIntroduceVariable(
+                bool isSpanEmpty,
                 CancellationToken cancellationToken)
             {
                 if (!_service.CanIntroduceVariableFor(this.Expression))
                 {
                     return false;
                 }
 
-                if (this.Expression is TTypeSyntax)
+                if (isSpanEmpty && this.Expression is TNameSyntax)
+                {
+                    // to extract a name, you must have a selection (this avoids making the refactoring too noisy)
+                    return false;
+                }
+
+                if (this.Expression is TTypeSyntax && !(this.Expression is TNameSyntax))
```

Comment:
probably good to comment this.

Reply:
so condition that can hit here is

1, span not empty and expression is not name syntax 
2, span is empty but expression is not name syntax 
3. span not empty but expression is name syntax 

so, if expression is not name syntax, regardless of span, if expression is typesyntax, we don't support refactoring.

shouldn't this check move before isSpanEmpty check?